include/seastar/testing/test_fixture.hh

```diff
@@ -0,0 +1,188 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * Copyright (C) 2025 ScyllaDB Ltd.
+ */
+
+#pragma once
+
+#include <functional>
+
+#include <seastar/core/future.hh> // IWYU pragma: keep
+#include <seastar/core/thread.hh>
+#include <seastar/util/defer.hh>
+#include <seastar/testing/seastar_test.hh>
+#include <seastar/testing/test_runner.hh>
+
+namespace seastar::testing {
+    namespace detail {
+        // could/should make futurize::invoke use std::invoke so
+        // it can handle member function, but don't want to change
+        // all code in the universe for this here.
```

Comment:
What are the differences between this and futurize_invoke()?`

Reply:
Fair enough. Done.